kubeadm/kube-proxy/hns.psm1

```diff
@@ -0,0 +1,36 @@
+Param(
+    [parameter(Mandatory = $true)] $NetworkName="flannel.4096"
+)
+
+# get surce vip
+function Get-SourceVip()
+{
+    $hnsNetwork = Get-HnsNetwork | ? Name -EQ $NetworkName.ToLower()
+    $subnet = $hnsNetwork.Subnets[0].AddressPrefix
+
+    $ipamConfig = @"
+{"cniVersion": "0.2.0", "name": "$NetworkName", "ipam":{"type":"host-local","ranges":[[{"subnet":"$subnet"}]],"dataDir":"/var/lib/cni/networks"}}
+"@
+
+    $ipamConfig | Out-File "c:\k\sourceVipRequest.json"
+
+    $env:CNI_COMMAND="ADD"
+    $env:CNI_CONTAINERID="dummy"
+    $env:CNI_NETNS="dummy"
+    $env:CNI_IFNAME="dummy"
+    $env:CNI_PATH="c:\opt\cni\bin" #path to host-local.exe
+
+    cd $env:CNI_PATH
+    If(!(Test-Path c:\k\sourceVip.json)){
+        Get-Content c:\k\sourceVipRequest.json | .\win-overlay.exe | Out-File "c:\k\sourceVip.json"
```

Comment:
I think this line needs to run .\host-local    win-overlay.exe doesn't understand any of the CNI env variables for dummy, and  the sourceVipRequest.json also has type=host-local

Reply:
Is host-local, negligence check